### PR TITLE
configure goreleaser to build and publish artifacts

### PR DESCRIPTION
**Pull request by fho**
Configure goreleleaser to build and publish release artifacts.

The process is documented in docs/RELEASE.md.

We could setup a github action workflow to automate it further: https://goreleaser.com/ci/actions/.

Please also review the created test release: https://github.com/simplesurance/dependencies-tool/releases/tag/untagged-3c59dd6454cbfbd7e902

### CODE REVIEW

File: docs/RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+6. Review the created draft release on
+   [GitHub](https://github.com/simplesurance/dependencies-tool/releases) and
+   publish it.
+   
+7. If the release was published, push the local git-tag:
+
+    ```sh
+    git push --tags
+    ```
+  If the release was discarded, delete the local git tag.
```
Comment (fho) [inline docs/RELEASE.md:46]: replace with a note that the tag does not need to be pushed and will be created when removing the draft status for the release on github

File: .goreleaser.yaml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    name_template: >-
+      {{ .ProjectName }}_
+      {{- title .Os }}_
+      {{- if eq .Arch "amd64" }}x86_64
+      {{- else if eq .Arch "386" }}i386
+      {{- else }}{{ .Arch }}{{ end }}
+      {{- if .Arm }}v{{ .Arm }}{{ end }}
+    # use zip for windows archives
+    format_overrides:
+      - goos: windows
```
Comment (FranciscoKurpiel) [inline .goreleaser.yaml:34]: On `builds.goos` you only list `linux` and `darwin`. Why is a directive here for windows?

**Review by nocive — APPROVED** [review]

File: .goreleaser.yaml (hunk 2 above)
Comment (fho) [reply .goreleaser.yaml:34]: thx, forgot to remove it here

Comment (fho) [reply .goreleaser.yaml]: fixed

File: docs/RELEASE.md (hunk 1 above)
Comment (fho) [reply docs/RELEASE.md]: fixed

**Pull request merged**